Replace empty shared pointer by default constructed DigitalReplication

pcaspers opened the pull request.

wegamekinglc commented:
I tried this on windows machine with vc11. It doesn't compile with boost 1.55. Compiler complains can't match any existing make_shared template...

pcaspers commented:
what if you replace " = make_shared ..." by " =
boost::shared_ptr<DigitalReplication>(new DigitalReplication())", would
that work ?

On 25 March 2015 at 07:41, Cheng Li <notifications@github.com> wrote:

> I tried this on windows machine with vc11. It doesn't compile with boost
> 1.55. Compiler complains can't match any existing make_shared template...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lballabio/quantlib/pull/207#issuecomment-85877368>.
>

wegamekinglc commented:
Hi Peter. Yes, that solution works.

pcaspers commented:
ok thanks, I will update the pr

Von meinem iPhone gesendet

> Am 01.04.2015 um 04:13 schrieb Cheng Li <notifications@github.com>:
> 
> Hi Peter. Yes, that solution works.
> 
> —
> Reply to this email directly or view it on GitHub.
>

pcaspers commented:
ok, it should work on msvc now. I also changed the digital cms spread coupon classes in this PR which I think are already merged into the master. Hope that works, since it looks as if I added new files here on github.
